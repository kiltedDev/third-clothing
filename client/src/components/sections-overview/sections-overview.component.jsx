import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { selectSectionsForPreview } from '../../redux/shop/shop.selectors';
import SectionPreview from '../section-preview/section-preview.component';

import { SectionsOverviewContainer } from './sections-overview.styles';

export const SectionsOverview = ({ sections }) => (
  <SectionsOverviewContainer className='sections-overview'>
    {sections.map(({ id, ...otherSectionProps }) => (
      <SectionPreview key={id} {...otherSectionProps} />
    ))}
  </SectionsOverviewContainer>
);

const mapStateToProps = createStructuredSelector({
  sections: selectSectionsForPreview
})

export default connect(mapStateToProps)(SectionsOverview);
