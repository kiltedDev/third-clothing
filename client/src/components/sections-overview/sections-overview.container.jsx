import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import { selectIsSectionFetching } from '../../redux/shop/shop.selectors';
import WithSpinner from '../with-spinner/with-spinner.component';
import SectionsOverview from './sections-overview.component';

const mapStateToProps = createStructuredSelector({
  isLoading: selectIsSectionFetching
});

const SectionsOverviewContainer = compose(
  connect(mapStateToProps),
  WithSpinner
)(SectionsOverview);

export default SectionsOverviewContainer;
