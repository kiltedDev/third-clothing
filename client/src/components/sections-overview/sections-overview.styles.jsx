import styled from 'styled-components';

export const SectionsOverviewContainer = styled.div`
  display: flex;
  flex-direction: column;
`

SectionsOverviewContainer.displayName = 'SectionsOverviewContainer';
