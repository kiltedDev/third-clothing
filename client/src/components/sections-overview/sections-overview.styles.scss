.sections-overview {
  display: flex;
  flex-direction: column;
}
