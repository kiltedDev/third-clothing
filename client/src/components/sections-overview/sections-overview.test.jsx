import { shallow } from 'enzyme';
import React from 'react';
import { SectionsOverview } from './sections-overview.component';

describe('SectionsOverview component', () => {
  let wrapper;
  let snapshot;
  beforeAll(() => {
    const mockProps = {
      sections: [
        {
          id: 1
        },
        {
          id: 2
        },
        {
          id: 3
        }
      ]
    };
    wrapper = shallow(<SectionsOverview { ...mockProps } />);
    snapshot = wrapper.debug();
  });

  it('should render SectionsOverview component',() => {
    expect( snapshot ).toMatchSnapshot();
  });
});
