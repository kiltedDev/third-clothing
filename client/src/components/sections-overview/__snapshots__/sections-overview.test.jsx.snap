// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionsOverview component should render SectionsOverview component 1`] = `
"<SectionsOverviewContainer className=\\"sections-overview\\">
  <SectionPreview />
  <SectionPreview />
  <SectionPreview />
</SectionsOverviewContainer>"
`;
